clang/lib/CodeGen/CGExprScalar.cpp

```diff
@@ -1097,6 +1112,27 @@ void ScalarExprEmitter::EmitIntegerTruncationCheck(Value *Src, QualType SrcType,
                 {Src, Dst});
 }
 
+static llvm::Value *EmitIsNegativeTestHelper(Value *V, QualType VType,
+                                             const char *Name,
+                                             CGBuilderTy &Builder) {
+  // NOTE: zero value is considered to be non-negative.
```

Comment:
This is self-evident and does not need to be in a comment.  Actually, most of the comments in this function are pretty unnecessary; you can just say something like "If the type is unsigned, the value is never negative." and leave everything else self-documenting.

Reply:
Leftover comment from previous patches, will update.